### PR TITLE
Return Backup and Recovery status and progress

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7255/app_12423_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7257/app_12423_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7258/app_12423_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7258/HighFidelity-Beta-PR12423-2b85634a21abec80f8f689013c2696afb19e2dc9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7258/HighFidelity-Beta-PR12423-2b85634a21abec80f8f689013c2696afb19e2dc9.exe

File: domain-server/src/DomainContentBackupManager.cpp [hunk 1]
```diff
@@ -280,12 +314,48 @@ std::vector<BackupItemInfo> DomainContentBackupManager::getAllBackups() {
...
+                { "createdAtMillis", createdAt.toMSecsSinceEpoch() },
+                { "isAvailable", isAvailable },
+                { "availabilityProgress", availabilityProgress },
+                { "isManualBackup", type == MANUAL_BACKUP_PREFIX }
+            }));
+        }
+    }
+
+    float recoveryProgress = 0.0f;
+    bool isRecovering = _isRecovering.load();
```
Comment (birarda) [inline domain-server/src/DomainContentBackupManager.cpp]: How often does process get called (how long could this be stale for)?

File: domain-server/src/DomainContentBackupManager.cpp [hunk 2]
```diff
@@ -383,32 +453,62 @@ void DomainContentBackupManager::backup() {
...
-        if (!zip.open(QuaZip::mdAdd)) {
-            qCritical() << "Could not open backup archive:" << filePath;
-            qCritical() << "    ERROR:" << zip.getZipError();
-            return;
-        }
+    {
+        QFile copyFile(copyFilePath);
+        copyFile.remove();
+        copyFile.close();
+    }
```
Comment (birarda) [inline domain-server/src/DomainContentBackupManager.cpp]: ?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7261/app_12423_PR.apk

File: domain-server/src/DomainContentBackupManager.cpp [hunk 3]
```diff
@@ -383,32 +453,62 @@ void DomainContentBackupManager::backup() {
...
+    zip.close();
 
-        zip.close();
+    if (zip.getZipError() != UNZ_OK) {
+        qCritical() << "Failed to consolidate backup: " << zip.getZipError();
+        promise->resolve({ { "success", false } });
+        return;
     }
+
+    qDebug() << "copyFilePath" << copyFilePath;
```
Comment (birarda) [inline domain-server/src/DomainContentBackupManager.cpp]: need this still?

File: domain-server/src/DomainServer.cpp [hunk 4]
```diff
@@ -2129,24 +2128,35 @@ bool DomainServer::handleHTTPRequest(HTTPConnection* connection, const QUrl& url
...
+            auto id = url.path().mid(QString(URI_API_BACKUPS_ID).length());
+            auto deferred = makePromise("consolidateBackup");
+            deferred->then([connection, JSON_MIME_TYPE](QString error, QVariantMap result) {
+                QJsonObject rootJSON;
+                auto success = result["success"].toBool();
+                if (success) {
+                    auto path = result["backupFilePath"].toString();
+                    auto file { std::unique_ptr<QFile>(new QFile(path)) };
+                    if (file->open(QIODevice::ReadOnly)) {
+                        connection->respond(HTTPConnection::StatusCode200, std::move(file));
```
Comment (birarda) [inline domain-server/src/DomainServer.cpp]: Do we need to clean up these files once the connection is done responding?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7261/HighFidelity-Beta-PR12423-697f0c443cb0f3606ca2facc53dc35bd5ca98ae0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7261/HighFidelity-Beta-PR12423-697f0c443cb0f3606ca2facc53dc35bd5ca98ae0.exe

File: domain-server/src/DomainContentBackupManager.cpp (hunk 2 above)
Comment (Atlante45) [reply domain-server/src/DomainContentBackupManager.cpp]: If the file already exists, the copy call will fail.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7262/app_12423_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7262/HighFidelity-Beta-PR12423-b30f98d5414759a1eae88505812cb481bb813844.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7262/HighFidelity-Beta-PR12423-b30f98d5414759a1eae88505812cb481bb813844.exe

**Pull request merged**